deckhouse-controller/pkg/controller/module-controllers/source/controller.go

```diff
@@ -278,6 +284,57 @@ func (c *Controller) createOrUpdateReconcile(ctx context.Context, roMS *v1alpha1
 	// Or create a copy manually for better performance
 	ms := roMS.DeepCopy()
 
+	// check if stored checksum equals to the checksum of the current config
+	if ms.Status.ConfigChecksum != controllerUtils.CalculateModuleSourceChecksum(&ms.Spec.Registry) {
```

Comment:
Why is a checksum required in the first place? Don't you know the registry configuration you injected into the open API?

Reply:
If you can get the current values for the module, you can avoid using checksums at all, which is a cleaner solution.

Reply:
As I understand, we have registry settings in two places: a module source object and its related module releases (openapi/values.yaml file). So, I decided to store a checksum so as to be able to detect a relevant change of a module source object.